homeassistant/components/recorder/core.py

```diff
@@ -741,52 +750,55 @@ def _run(self) -> None:
             self.migration_is_live = migration.live_migration(schema_status)
 
         self.hass.add_job(self.async_connection_success)
-        database_was_ready = self.migration_is_live or schema_status.valid
-
-        if database_was_ready:
-            # If the migrate is live or the schema is valid, we need to
-            # wait for startup to complete. If its not live, we need to continue
-            # on.
-            self._activate_and_set_db_ready()
-
-            # We wait to start a live migration until startup has finished
-            # since it can be cpu intensive and we do not want it to compete
-            # with startup which is also cpu intensive
-            if self._wait_startup_or_shutdown() is SHUTDOWN_TASK:
-                # Shutdown happened before Home Assistant finished starting
+
+        # First do non-live migration steps, if needed
+        if not schema_status.valid:
+            result, schema_status = self._migrate_schema_offline(schema_status)
+            if not result:
+                self._notify_migration_failed()
                 self.migration_in_progress = False
-                # Make sure we cleanly close the run if
-                # we restart before startup finishes
                 return
+            self.schema_version = schema_status.current_version
+            self.migration_is_live = True
```

Comment:
This is confusing that we set `self.migration_is_live = True` after an offline migration.  Would you please add a comment here

Reply:
Fixed